src/aux-server/server/modules/GpioModule3.ts

```diff
@@ -0,0 +1,491 @@
+const SerialPort = require('serialport');
+const parsers = SerialPort.parsers;
+
+let btSerial = new Map<string, typeof SerialPort>();
+
+/**
+ * Defines an AuxModule that adds Serial functionality to the module.
+ */
+export class GpioModule3 implements AuxModule2 {
+    constructor() {}
+
+    setChannelManager() {}
+
+    async setup(simulation: Simulation): Promise<Subscription> {
+        let sub = new Subscription();
+
+        sub.add(
+            simulation.localEvents
+                .pipe(
+                    flatMap(async event => {
+                        if (event.type === 'serial_connect') {
+                            await this._serialConnect(simulation, event);
+                        }
+                        if (event.type === 'serial_stream') {
+                            await this._serialStream(simulation, event);
+                        }
+                        if (event.type === 'serial_open') {
+                            await this._serialOpen(simulation, event);
+                        }
+                        if (event.type === 'serial_update') {
+                            await this._serialUpdate(simulation, event);
+                        }
+                        if (event.type === 'serial_write') {
+                            await this._serialWrite(simulation, event);
+                        }
+                        if (event.type === 'serial_read') {
+                            await this._serialRead(simulation, event);
+                        }
+                        if (event.type === 'serial_close') {
+                            await this._serialClose(simulation, event);
+                        }
+                        // if (event.type === 'serial_set') {
+                        //     await this._serialSet(simulation, event);
+                        // }
+                        // if (event.type === 'serial_get') {
+                        //     await this._serialGet(simulation, event);
+                        // }
+                        // if (event.type === 'serial_flush') {
+                        //     await this._serialFlush(simulation, event);
+                        // }
+                        // if (event.type === 'serial_drain') {
+                        //     await this._serialDrain(simulation, event);
+                        // }
+                        if (event.type === 'serial_pause') {
+                            await this._serialPause(simulation, event);
+                        }
+                        if (event.type === 'serial_resume') {
+                            await this._serialResume(simulation, event);
+                        }
+                    })
+                )
+                .subscribe()
+        );
+        return sub;
+    }
+
+    _serialConnect(simulation: Simulation, event: SerialConnectAction) {
+        try {
+            // TODO: Pass the device name through the webhook?
```

Comment:
@WesleyMayle we should note in the documentation that only the toothbrush devices are supported currently because that is the only device you can connect to via the API at the moment.

Reply:
The casualOS side of this actually supports anything connecting to rfcomm0, which should be all bluetooth serial connections. Aux-cli though, is currently hardcoded for toothbrush stuff.

Reply:
casualOS only supports one device at a time I guess though.

Reply:
yeah until we make it possible to connect to something else via aux-cli then we should document that limitation